### PR TITLE
LLM: fix input length logic for run_transformer_int4_gpu

### PR DESCRIPTION
**Pull request by WeiguangHan**
@hkvision Fix input length logic for run_transformer_int4_gpu.

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -399,8 +399,10 @@ def run_transformer_int4_gpu(repo_id,
             # in_len.txt maybe shorter than we need,
             # use much longer context to make sure input length
             test_length = min(in_len*2, 8192)
-            while test_length not in [32, 256, 1024, 2048, 8192]:
+            prompt_list = [32, 256, 1024, 2048, 8192]
+            while test_length not in prompt_list and test_length < max(prompt_list):
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:403]: just `while test_length not in [32, 256, 1024, 2048, 8192] and test_length < 8192:`?

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 2]
```diff
@@ -399,8 +399,10 @@ def run_transformer_int4_gpu(repo_id,
...
                 test_length = test_length * 2
+            test_length = min(test_length, 8192)
```
[the remark this answers is not in the document]
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:405]: sure

File: python/llm/dev/benchmark/all-in-one/run.py (hunk 1 above)
Comment (WeiguangHan) [reply python/llm/dev/benchmark/all-in-one/run.py:403]: ok

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 3]
```diff
@@ -399,9 +399,9 @@ def run_transformer_int4_gpu(repo_id,
             # in_len.txt maybe shorter than we need,
             # use much longer context to make sure input length
             test_length = min(in_len*2, 8192)
-            prompt_list = [32, 256, 1024, 2048, 8192]
-            while test_length not in prompt_list and test_length < max(prompt_list):
+            while test_length not in [32, 256, 1024, 2048, 8192] and test_length < 8192:
                 test_length = test_length * 2
+            # For the sequence length not in [32, 256, 1024, 2048, 8192], they will be truncated from 8192.txt.
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:404]: they -> it

**Review by hkvision — APPROVED** [review]

**Pull request merged**